python/paddle/fluid/tests/unittests/mkldnn/test_conv2d_mkldnn_op.py

```diff
@@ -15,44 +15,158 @@
+                output = output.astype(self.dtype)
+                self.inputs['Bias'] = OpTest.np_dtype_to_fluid_dtype(bias)
+
+        if hasattr(self, "fuse_residual"):
+            if self.fuse_residual and hasattr(self, "input_residual_size"):
+                input_residual = np.random.random(
+                    self.input_residual_size).astype(self.dtype)
+                output = conv_2d_residual_naive(output, input_residual)
+                output = output.astype(self.dtype)
+
+                self.attrs['fuse_residual_connection'] = self.fuse_residual
+                self.inputs['ResidualData'] = OpTest.np_dtype_to_fluid_dtype(
+                    input_residual)
+
+        self.outputs['Output'] = output
+
+
+class TestMKLDNNFuseReluBiasResidual(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        # TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.fuse_bias = True
+        self.fuse_residual = True
+        self.fuse_relu_before_depthwise_conv = True
+        self.bias_size = [6]
+        self.input_residual_size = [2, 6, 5, 5]
+        self.pad = [1, 1]
+        self.stride = [1, 1]
+        self.input_size = [2, 3, 5, 5]  # NCHW
+        assert np.mod(self.input_size[1], self.groups) == 0
+        f_c = self.input_size[1] // self.groups
+        self.filter_size = [6, f_c, 3, 3]
+
+    def test_check_grad(self):
+        pass
+
+    def test_check_grad_no_filter(self):
+        pass
+
+    def test_check_grad_no_input(self):
+        pass
+
+
+class TestWithPadWithBias(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.pad = [1, 1]
+        self.fuse_bias = True
+        self.bias_size = [6]
+        self.fuse_residual = True
+        self.input_residual_size = [2, 6, 5, 5]
+
+    def test_check_grad(self):
+        pass
+
+    def test_check_grad_no_filter(self):
+        pass
+
+    def test_check_grad_no_input(self):
+        pass
+
+
+class TestWithStride(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.pad = [1, 1]
+        self.stride = [2, 2]
+        self.input_size = [2, 3, 6, 6]
+
+
+class TestWithGroup(TestConv2dMKLDNNBiasResidualOp):
+    def init_group(self):
+        self.groups = 3
+
+
+class TestWith1x1(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.input_size = [2, 3, 5, 5]  # NCHW
+        assert np.mod(self.input_size[1], self.groups) == 0
+        f_c = self.input_size[1] // self.groups
+        self.filter_size = [6, f_c, 1, 1]
+
+    def init_group(self):
+        self.groups = 3
+
+
+class TestWithInput1x1Filter1x1(TestConv2dMKLDNNBiasResidualOp):
+    def init_test_case(self):
+        TestConv2dMKLDNNBiasResidualOp.init_test_case(self)
+        self.input_size = [2, 3, 1, 1]  # NCHW
```

Comment:
Consider removing all the following lines of this method since they already repeat what has been initialised in the parent method.